dspace-server-webapp/src/test/java/org/dspace/app/rest/WorkspaceItemRestRepositoryIT.java

```diff
@@ -2444,7 +2452,8 @@ public void patchUploadTest() throws Exception {
         // check that changes persist
         getClient(authToken).perform(get("/api/submission/workspaceitems/" + witem.getID()))
             .andExpect(status().isOk())
-            .andExpect(jsonPath("$.sections.upload.files[0].metadata['dc.source']").doesNotExist())
+            .andExpect(jsonPath("$.sections.upload.files[0].metadata['dc.source'][0].value",
+                    is("/local/path/simple-article.pdf")))
```

Comment:
Is there a reason we switched this test to no longer remove `dc.source` and instead just verify it exists?  Just curious why we switched this test.

Reply:
this metadata is not included in the form definition https://github.com/DSpace/DSpace/blob/master/dspace/config/submission-forms.xml#L24 so as this PR has fixed the patch request to respect the configuration preventing to update not configured metadata the dc.source should remain untouched.

One related question/issue is: should we update also the GET method to respect the configuration exposing only the configured metadata? the rest contract seems to imply that
https://github.com/DSpace/Rest7Contract/blob/master/workspaceitem-data-upload.md

in this case the test should be changed to verify the dc.source using a separate call to the bitstream endpoint. It is a bit out-of-scope of this PR but if we agree it could be easier to just apply the change also to the https://github.com/DSpace/DSpace/blob/8c234d1a5c574c17d1c2687e2121bd8b64d9820b/dspace-server-webapp/src/main/java/org/dspace/app/rest/submit/SubmissionService.java#L154

Reply:
Thanks for the explanation. I'm not sure about updating the `GET`, as that seems to imply that some metadata could be "hidden" (filtered out) in the `GET` if it's not in the current submission configuration. That seems like it could have side effects if some metadata if filtered out so that it is "invisible" to the client side.   In any case, I agree it should not be part of this PR & I think it needs more analysis.

For the purposes of this PR, your initial answer is good enough here. The `dc.source` is not in the default configuration, so it cannot be PATCHed.